common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -420,18 +432,88 @@ private void notifyListeners() {
         }
         EventExecutor executor = executor();
         if (executor.inEventLoop()) {
```

Comment:
> Is there concern that the executor may run the OneTimeTask immediately (not enqueuing it)?

Yes. For example `ImmediateExecutor` (and executors which behave in a similar fashion) would not work with the "run a task on the executor" approach.